test/e2e/framework/network/utils.go

```diff
@@ -93,15 +93,46 @@ var NetexecImageName = imageutils.GetE2EImage(imageutils.Agnhost)
 
 // NewNetworkingTestConfig creates and sets up a new test config helper.
 func NewNetworkingTestConfig(f *framework.Framework, hostNetwork, SCTPEnabled bool) *NetworkingTestConfig {
-	config := &NetworkingTestConfig{f: f, Namespace: f.Namespace.Name, HostNetwork: hostNetwork, SCTPEnabled: SCTPEnabled}
+	config := &NetworkingTestConfig{f: f,
+		Namespace:           f.Namespace.Name,
+		HostNetwork:         hostNetwork,
+		SCTPEnabled:         SCTPEnabled,
+		EndpointPodHTTPPort: EndpointHTTPPort,
+		EndpointPodUDPPort:  EndpointUDPPort,
+		EndpointPodSCTPPort: EndpointSCTPPort,
+	}
 	ginkgo.By(fmt.Sprintf("Performing setup for networking test in namespace %v", config.Namespace))
 	config.setup(getServiceSelector())
 	return config
 }
 
+// NewHostNetworkingTestConfig creates and sets up a new test config helper using backend pods using host network.
+// we need to set an offset because the tests are run in parallel and, since the pods are using the hostNetwork,
+// they can not expose the same port or they will have a conflict and will not be scheduled
+func NewHostNetworkingTestConfig(f *framework.Framework, hostNetwork bool, portOffset int) *NetworkingTestConfig {
+	config := &NetworkingTestConfig{f: f,
+		Namespace:           f.Namespace.Name,
+		HostNetwork:         hostNetwork,
+		NetProxyHostNetwork: true,
```

Comment:
The old `HostNetwork`/`hostNetwork` is now confusing. And "proxy" to me sounds like it means "in the middle", but `NetProxyHostNetwork` isn't whether the "middle" pod is hostnetwork but rather whether the final pod is hostnetwork.

I'd call the old field `HostNetworkProxy` or `HostNetworkIntermediary` and the new one `HostNetworkDestination`.

And maybe it would be better to have just

    func NewNetworkingTestConfig(f *framework.Framework, options *NetworkingTestConfig) *NetworkingTestConfig

and the caller can pass in a partially-constructed test config with `HostNetworkProxy`, `HostNetworkDestination`, and `SCTPEnabled` set if they need those options?

And get rid of the `portOffset` arg; have `NewNetworkingTestConfig` set it itself when using `HostNetworkDestination`. Just have a global (and a mutex protecting it) and take the next available offset each time it's called.

Reply:
> and the caller can pass in a partially-constructed test config with HostNetworkProxy, HostNetworkDestination, and SCTPEnabled set if they need those options?

Better using functional options since it seems the idiomatic way and is allows to add new options easily

> The old HostNetwork/hostNetwork is now confusing. And "proxy" to me sounds like it means "in the middle", but NetProxyHostNetwork isn't whether the "middle" pod is hostnetwork but rather whether the final pod is hostnetwork.

absolutely agree, very bad naming, 
`HostNetwork` remains the same, for endpoints with `HostNetwork` the new variable is `EndpointsHostNetwork` now, and they have independent setters

> And get rid of the portOffset arg; have NewNetworkingTestConfig set it itself when using HostNetworkDestination. Just have a global (and a mutex protecting it) and take the next available offset each time it's called.

yeah, the portOffset is wrong but I prefer to avoid to use a global because it will make the test port unpredictable and the randomness cause lot of troubles to debug flakes and failures, also, I'm not sure how ginkgo parallel will work with that approach (see the `Parallel Specs` section in https://onsi.github.io/ginkgo/)

I prefer to keep it simple,  I've added setters so we can configure the endpoints ports in each test, and is not that we expect more changes on this part in the future.